src/uct/ib/base/ib_md.c

```diff
@@ -814,17 +880,23 @@ static ucs_status_t uct_ib_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
     uct_ib_mem_t *memh      = uct_memh;
     uint32_t atomic_rkey;
     ucs_status_t status;
+    int mr_index;
 
     /* create umr only if a user requested atomic access to the
      * memory region and the hardware supports it.
      */
-    if ((memh->flags & UCT_IB_MEM_ACCESS_REMOTE_ATOMIC) &&
-        !(memh->flags & UCT_IB_MEM_FLAG_ATOMIC_MR) &&
+    if ((((memh->flags & UCT_IB_MEM_ACCESS_REMOTE_ATOMIC) &&
+        !(memh->flags & UCT_IB_MEM_FLAG_ATOMIC_MR)) || md->relaxed_order) &&
         (memh != md->global_odp))
     {
+        if (memh->flags & UCT_IB_MEM_FLAG_RELAXED_ORDERING) {
```

Comment:
need to check relaxed order flag on the MR?

Reply:
no

Reply:
maybe in line 889 we can check (memh->flags & UCT_IB_MEM_FLAG_RELAXED_ORDERING) instead of md->relaxed_order? instead of line 892?
since if md->relaxed_order is true, we always set UCT_IB_MEM_FLAG_RELAXED_ORDERING